lib/luacrypto_skcipher.c

```diff
@@ -0,0 +1,254 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for
+* symmetric-key ciphers (SKCIPHER).
+*
+* This module provides a `new` function to create SKCIPHER transform objects,
+* which can then be used for encryption and decryption with various block cipher
+* algorithms and modes.
+*
+* @module crypto_skcipher
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/skcipher.h>
+#include <linux/err.h>
+#include <linux/scatterlist.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+static const lunatik_class_t luacrypto_skcipher_class;
+
+typedef struct {
+	struct crypto_skcipher *tfm;
+	struct skcipher_request *req;
```

Comment:
is the idea to have a single request per object? wondering if it wouldn't be the case to allocate it per request.. what's the common pattern in the kernel for using this API? can you describe pros and cons of allocating it per object instead of per request call? is it safe to reuse this structure on multiple requests?